homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -0,0 +1,100 @@
+"""The Husqvarna Automower integration."""
+from asyncio.exceptions import TimeoutError as AsyncioTimeoutError
+import contextlib
+import logging
+from typing import cast
+
+import aioautomower
+
+from homeassistant.components.application_credentials import DATA_STORAGE
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AutomowerDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching Husqvarna data."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize data updater."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+        )
+        api_key = None
+        ap_storage: dict = cast(dict, hass.data.get("application_credentials"))
+        ap_storage_data: dict = ap_storage[DATA_STORAGE].__dict__["data"]
+        for k in ap_storage_data:
+            api_key = ap_storage_data[k]["client_id"]
+        entry_dict = entry.as_dict()
+        access_token = entry_dict["data"]["token"]
+        scope = entry_dict["data"]["token"]["scope"]
```

Comment:
I am not sure how at this point, but want to mention it, I think this is data that should be in the entry itself, I'll have a second look later.

Reply:
I had this in the beginning:
```
    implementation = await async_get_config_entry_implementation(hass, entry)
    coordinator = AutomowerDataUpdateCoordinator(
        hass,
        implementation,
        entry=entry,
    )
```
and used in the coordinator `api_key = implementation.client_id`. But the I got problems with the tests I couldn't solved.

Reply:
We will take a look at this when we start with the tests